### PR TITLE
[WIP] add a basic dockerfile that pulls in all of the local content and run quickstart

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.cfme_venv/
+build/
+dist/
+docs/_build
+*.pyc
```
Comment (bsquizz) [inline .dockerignore]: Some more dirs to add here:
.cache/
.pytest_cache/
.idea/ (for those on pycharm :P)

And not sure what we want to do about .yaml_key -- we could copy it in as long as we're not releasing the built image with the yaml key publicly

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+RUN python -m cfme.scripting.quickstart --mk-virtualenv /cfme_venf
+
+
+
+RUN dnf install -y git
+RUN git config --global user.name DockerCfme
+run git config --global user.email no-reply@unused.example.redhat.com
+
+
+COPY . .
```
Comment (bsquizz) [inline Dockerfile]: Why do we do the copies above and then "COPY ." here? Wouldn't this take care of copying all the same files?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+COPY cfme/__init__.py cfme/__init__.py
+COPY requirements requirements
+COPY setup.py setup.py
+RUN mkdir conf
+RUN python -m cfme.scripting.quickstart --mk-virtualenv /cfme_venf
+
+
+
+RUN dnf install -y git
+RUN git config --global user.name DockerCfme
```
Comment (bsquizz) [inline Dockerfile]: Do we need to do anything more to blow away all the git config? Like rm .git/config? Or will this do?

**Comment by bsquizz** [comment]
I'd suggest adding the VENV to the path so we can run miq commands straight from the 'docker run', with a default command of 'miq shell'
For example:
```
ENV PATH="/cfme_venf/bin:${PATH}"
CMD ["miq", "shell"]
```

File: Dockerfile (hunk 3 above)
Comment (RonnyPfannschmidt) [reply Dockerfile]: this just ensures git works

File: Dockerfile (hunk 2 above)
Comment (RonnyPfannschmidt) [reply Dockerfile]: better caching of lower layers

**Comment by RonnyPfannschmidt** [comment]
venv will explicitly not added to the path to avoid certain pathological effects

**Comment by RonnyPfannschmidt** [comment]
abadonded

**Pull request closed without merging**